src/components/NotificationIcon/NotificationIcon.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { INotificationIconProps, DefaultIconNumberStyleObject, NotificationBubbleStyleObject} from './NotificationIcon.Props';
+import { IconSize } from '../Icon/Icon.Props';
+import { getNativeAttributes, htmlElementAttributes } from '../../utilities/attributes';
+
+export const NotificationIcon: (props: INotificationIconProps) => JSX.Element = (props: INotificationIconProps) => {
+    const customIcon = props.iconName === '';
+    const svgIcon = props.iconName && props.iconName.startsWith('svg');
+    let iconPrefix = 'icon '.concat(svgIcon ? 'svg' : 'font');
+    let iconClassName = classNames(
+        [iconPrefix], {
+            [props.iconName]: !customIcon
+        }, [props.className]);
+
+    let iconHeight: any;
+    let iconWidth: any;
+
+    switch (props.iconSize) {
+        case IconSize.smallest:
+            iconHeight = '16px';
+            iconWidth = '16px';
+            break;
+        case IconSize.small:
+            iconHeight = '24px';
+            iconWidth = '24px';
+            break;
+        case IconSize.medium:
+            iconHeight = '32px';
+            iconWidth = '32px';
+            break;
+        case IconSize.large:
+            iconHeight = '64px';
+            iconWidth = '64px';
+            break;
+        default:
+        iconHeight = props.height || '16px';
+        iconWidth = props.width || '16px';
+    }
+
+    const numberString = props.notificationNumber === undefined || props.notificationNumber === 0 ? '' 
+                            : props.notificationNumber < 10 ? props.notificationNumber.toLocaleString() : '9+';
+    const defaultStyleObject = DefaultIconNumberStyleObject(props.iconSize, numberString.length);
+   
+    let styleObject: NotificationBubbleStyleObject = { ...defaultStyleObject};
```

Comment:
down below on line 49. will actually also modify the defaultStyleObject since the containerStyleObject of styleObject is te same reference if you think about it. so if we are doing it like this then the spread operator is not even necessary since the defaultStyleObject will be mutated anyways. So there are two choices.
1. rename the DefaultIconNumberStyleObject function to CreateDefaultIconNumberStyleObject so that it is evident that mutation was intended
2. spread all of the subobjects to a different reference will be used below when using object.assign